### DISCUSSION TITLE
Empty states of widgets

### DISCUSSION THREAD

**Discussion by juliangums** [filamentphp/filament#13311] [Help] [answered]
### Package

Widgets

### Package Version

latest

### How can we help you?

I have been able to build some nice dynamically filters charts like this one:
<img width="1270" alt="Screenshot 2024-06-18 at 20 00 58" src="https://github.com/filamentphp/filament/assets/22907521/c5b9785a-acf5-4914-9e1f-5a69a101d3df">

However, if there is no data, which will definitely happen on new setups and tight filtering, it just looks like this:
<img width="1254" alt="Screenshot 2024-06-18 at 20 02 55" src="https://github.com/filamentphp/filament/assets/22907521/8222f01a-568b-4447-afc0-fb2c948109e3">

That kinda sucks. Is there an easy way to get the pretty empty states we have on tables?
<img width="1247" alt="Screenshot 2024-06-18 at 20 04 38" src="https://github.com/filamentphp/filament/assets/22907521/896e32ea-b95f-4eb2-9919-917dea9f0fcc">

**Comment by juliangums** [comment]
@danharrin any thoughts on this?

**Reply by krekas** [reply]
make a pr with your suggested solution...

**Reply by danharrin** [reply]
It will need to be opt in where you tell Filament what "empty" actually means regarding your data, some people might actually want an empty chart

**Reply by danharrin** [reply]
`public function isEmpty(): bool`

**Reply by juliangums** [reply]
Agreed. I'll try to work on this in the coming week.

**Reply by juliangums** [reply, marked as answer]
https://github.com/filamentphp/filament/pull/13408

**Reply by corgalore** [reply]
Any progress on this addition to Chart widgets? I though it was a good idea as well.
